src/xdist/dsession.py

```diff
@@ -205,6 +179,64 @@ def loop_once(self) -> None:
         if self.sched.tests_finished:
             self.triggershutdown()
 
+
+    def is_node_finishing(self, node: WorkerController):
+        """Check if a test worker is considered to be finishing.
+
+        Evaluate whether it's on its last test, or if no tests are pending.
+        """
+        pending = self.sched.node2pending.get(node)
+        return pending is not None and len(pending) < 2
+
+
+    def is_node_clear(self, node: WorkerController):
```

Comment:
This looks unused

Reply:
It is unused, I mostly used it while debugging. I can remove it.